### PR TITLE
Add init. impl. of `NumberBaseConversion`

### LINKED ISSUE #61: Number base converter
A tool allowing easy convertion between hex, dec and oct number bases.

### PR DESCRIPTION
**Pull request by 7sne**
Closes #61

### CODE REVIEW

**Comment by LogvinovLeon** [comment]
There is a bug with the octal numbers. When you start typing decimal—it converts and puts no 0 prefix at the beginning. But when you put an octal number without a prefix—you get an error. I think it should either accept one of those formats or both, but accepting one on read and the other on write seems weird

**Review by LogvinovLeon — CHANGES_REQUESTED** [review]

File: src/lib/convertNumberBase.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import BigNumber from 'bignumber.js';
+
+import { NumberBaseType } from './numberBaseConvertProperties';
+
+export function convertNumberBase(
```
[the remark this answers is not in the document]
Comment (7sne) [reply src/lib/convertNumberBase.ts:5]: I thought about this approach. Although it's definitely smarter, I had a concern about readability. The current version is really self-explaining.

File: src/lib/convertNumberBase.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  from: number,
+  to: number,
+): string {
+  if (toBaseName === 'octal') {
+    return '0' + new BigNumber(value, from).toString(to);
+  } else if (toBaseName === 'hexadecimal') {
+    return '0x' + new BigNumber(value, from).toString(to);
+  } else {
+    return new BigNumber(value, from).toString(to);
+  }
```
Comment (LogvinovLeon) [inline src/lib/convertNumberBase.ts:51]:
```suggestion
  const numberPrefixes = {
    octal: '0',
    hexadecimal: '0x',
  };

  return (numberPrefixes[toBaseName] || '') + new BigNumber(value, from).toString(to);
```

File: src/lib/convertNumberBase.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            toBaseNumber,
+          );
+        }
+      }
+    }
+  }
+}
+
+// @internal
+const numberBaseConversionDict = {
```
Comment (LogvinovLeon) [inline src/lib/convertNumberBase.ts:31]: I don't like adding Dict at the end of the variable names and we've never done that before. This implies that we would also have XStr, XNumber, XList etc. We've had a convention for naming dicts. Either use some descriptive name or XToY or ByX. `baseToRadix` will do in that case.

File: src/lib/convertNumberBase.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    if (fromBaseName === fromUnit) {
+      for (const [toBaseName, toBaseNumber] of Object.entries(
+        numberBaseConversionDict,
+      )) {
+        if (toBaseName === toUnit) {
+          return convertNumberBaseWithBn(
+            value,
+            toBaseName,
+            fromBaseNumber,
+            toBaseNumber,
```
Comment (LogvinovLeon) [inline src/lib/convertNumberBase.ts:22]:
```suggestion
            fromRadix,
            toRadix,
```
1.
MATHEMATICS
the base of a system of numeration.

**Review by LogvinovLeon — CHANGES_REQUESTED** [review]

File: src/lib/convertNumberBase.ts (hunk 4 above)
Comment (7sne) [reply src/lib/convertNumberBase.ts:22]: but "the radix or **base** is the number of unique" :c

**Review by LogvinovLeon — APPROVED** [review]

**Pull request merged**